src/component/handlers/edit/editOnCompositionStart.js

```diff
@@ -18,9 +18,10 @@ var EditorState = require('EditorState');
  * The user has begun using an IME input system. Switching to `composite` mode
  * allows handling composition input and disables other edit behavior.
  */
-function editOnCompositionStart(): void {
+function editOnCompositionStart(e): void {
   this.setRenderGuard();
   this.setMode('composite');
+  this._onCompositionStart(e);
```

Comment:
Compositionstart can have a non-empty data event which indicate the part of text that is going to be replaced by composition. (on most major web browser). 

One alternative solution would be to stop caring about those events and diff the DOM at strategical points, updating the model based on content and not on half broken events. Not sure if it's easily doable.